docs/development/releasing.rst

```diff
@@ -45,22 +45,67 @@ packages that use the full bugfix/maintenance branch approach.)
    re-assign them to the next relevant milestone.
 
 #. (Only for major versions) Make sure to update the "What's new"
-   section with the stats on the number of issues, PRs, and contributors.  For
-   the first two, the `astropy-procedures repository`_ script ``gh_issuereport.py``
-   can provide the numbers since the last major release.  For the final one, you
-   will likely need to update the Astropy ``.mailmap`` file, as there are often
-   contributors who are not careful about using the same e-mail address for
-   every commit.  The easiest way to do this is to run the command
-   ``git shortlog -n -s -e`` to see the list of all contributors and their email
-   addresses.  Look for any misnamed entries or duplicates, and add them to the
-   ``.mailmap`` file (matched to the appropriate canonical name/email address.)
-   Once you have finished this, you can count the number of lines in
-   ``git shortlog -s`` to get the final contributor count.
-
-#. Also be sure to update the ``docs/credits.rst`` file to include any new
-   contributors from the above step. (This
-   step is only required on major releases, but can be done for bugfix releases
-   as time allows.)
+   section with the stats on the number of issues, PRs, and contributors. To do
+   this, use the `generate_releaserst.xsh`_ script. This requires `xonsh
+   <https://xon.sh/>`_ and `docopt <http://docopt.org/>`_ which you can install
+   with::
+
+      pip install xonsh docopt
+
+   You should then run the script in the root of the astropy repository as follows::
+
+      xonsh generate_releaserst.xsh 4.3 v4.3.dev \
+                                    --project-name=astropy \
+                                    --pretty-project-name=astropy \
+                                    --pat=<a GitHub personal access token>
+
+   The first argument should be the last major version (before any bug fix
+   releases, while the second argument should be the ``.dev`` tag that was just
+   after the branching of the last major version. Finally, you will need a
+   GitHub personal access token with default permissions (no scopes selected).
+
+   The output will look similar to::
+
+      This release of astropy contains 2573 commits in 163 merged pull requests
+      closing 104 issues from 98 people, 50 of which are first-time contributors
+      to astropy.
+
+      * 2573 commits have been added since 4.3
+      * 104 issues have been closed since 4.3
+      * 163 pull requests have been merged since 4.3
+      * 98 people have contributed since 4.3
+      * 50 of which are new contributors
+
+      The people who have contributed to the code for this release are:
+
+      - Name 1 *
+      - Name 2 *
+      - Name 3
+
+   At this point, you will likely need to update the Astropy ``.mailmap`` file,
```

Comment:
There is a pre-commit hook we could use that checks the mailmap (https://github.com/jumanjihouse/pre-commit-hooks#check-mailmap). The drawback is it puts the responsibility to edit the mailmap on first time contributors. Perhaps there's a way to auto-update the mailmap with the hook...

Reply:
It would be good to move towards some kind of automation for the mailmap though I agree that putting the burden on first time contributors is not ideal. But would be good to think about.